### PR TITLE
New PCL Alignment Threshold object in alignment consumers

### PR DESCRIPTION
**Pull request by dmeuser**
#### PR description:
This PR is the second step towards using a higher granularity (HG) for the pixel alignment in the PCL, following https://github.com/cms-sw/cmssw/pull/38195, where the new alignment threshold object was introduced. In this PR the consumer in the `Alignment` package are adapted to use the new threshold object.

In detail the following changes have been made:
- Change from `AlignPCLThresholds` to `AlignPCLThresholdsHG` in the consumer code of the `Alignment` package
- This PR updates the autoCond with GTs containing `SiPixelAliThresholdsHG_offline_v0` and `SiPixelAliThresholdsHG_express_v0`

#### PR validation:
The PR was tested and validated using `runTheMatrix -l 1001.0`

#### Upcoming PRs:
This PR will be followed by one additional PR, which will introduces the changes needed to run and store the HG alignment

#### New GlobalTags:
The new GTs are

124X_dataRun2_v2
124X_dataRun2_relval_v2
124X_dataRun2_HEfail_v2

124X_dataRun3_v3
124X_dataRun3_relval_v3

124X_dataRun3_Express_frozen_v1
124X_dataRun2_PromptLike_HI_v1
124X_dataRun3_Prompt_frozen_v1
where the frozen GTs are the snapshotter versions of the newly created

124X_dataRun3_Express_v1
124X_dataRun3_Prompt_v1
The snapshot time is
2022-06-09 20:00:00 (UTC)

The diffs

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_v1/124X_dataRun2_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_relval_v1/124X_dataRun2_relval_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_HEfail_v1/124X_dataRun2_HEfail_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_v2/124X_dataRun3_v3

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_relval_v2/124X_dataRun3_relval_v3

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun3_Express_frozen_v4/124X_dataRun3_Express_frozen_v1

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun2_PromptLike_HI_v3/124X_dataRun2_PromptLike_HI_v1

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun3_Prompt_frozen_v4/124X_dataRun3_Prompt_frozen_v1

cc:
@mmusich, @connorpa, @antoniovagnerini, @consuegs

### CODE REVIEW

**Comment by tvami** [comment]
type trk

**Comment by tvami** [comment]
> Add code to supply the AlignPCLThresholdsHGRcd via ESSource

Ok, let's test this as it is now, and then we should provide this Rcd from the GT in a follow-up. This we can do after the L1T changes have been in the GT (so 2-3 more days)

**Comment by tvami** [comment]
test parameters:
  - workflows = 1001.0

**Comment by tvami** [comment]
>  A new pre-release is needed for inclusion in GT (the record is currently not recognized by the browser)

I think that's not true. I can add it with admin rights, this is what I did 
AlignPCLThresholdsHGRcd | AlignPCLThresholdsHG
in the record management.

Please confirm that's the correct matching. 
If it is please upload the tag to Prod. The Prep DB is buggy so I couldnt do the same there

**Comment by mmusich** [comment]
>  would you mind expanding on why this is an unwelcome feature?

because it can lead the users (and the conveners unintentionally) to create broken global tags for the intended target queue.

**Comment by tvami** [comment]
OK the pre-release comes out today, so we can wait with this until tomorrow.

**Comment by mmusich** [comment]
@cms-sw/alca-l2 I see that the [12_5_0_pre2 build](https://github.com/cms-sw/cmssw/issues/38290) is ready. 
Shall we wait that it becomes propagated to the CondDB tools, re-upload, make new GTs, etc, or can we just proceed with this as is and fix it later? This PR is a blocker for the further (and largest) new developments concerning the HG Alignment PCL.

**Comment by mmusich** [comment]
>  Just a confirmation: this development is not meant to be for 12_3_X (just 12_4_X), correct?

it is meant for at least 12_4_X

**Comment by mmusich** [comment]
@tvami 

https://github.com/cms-sw/cmssw/pull/38297#issuecomment-1151043411 

has absolutely nothing to do with this. I hope you appreciate.

**Comment by mmusich** [comment]
please test

**Comment by tvami** [comment]
Hi Marco, I created the new GTs and announce them here
https://cms-talk.web.cern.ch/t/gt-new-pcl-alignment-threshold-object-in-alignment-consumers/11468/2
can you please include them in autoCond?

Please also update the comments where we say "identical to .... but with snapshot at 2022-06-09 20:00:00 (UTC)"

**Comment by mmusich** [comment]
@tvami 

> Hi Marco, I created the new GTs and announce them here
https://cms-talk.web.cern.ch/t/gt-new-pcl-alignment-threshold-object-in-alignment-consumers/11468/2
can you please include them in autoCond?

done.

> Please also update the comments where we say "identical to .... but with snapshot at 2022-06-09 20:00:00 (UTC)"

I don't own the PR thread, so @dmeuser must do that.

**Comment by tvami** [comment]
> I don't own the PR thread, so @dmeuser must do that.

I meant in the autoCond file

File: Configuration/AlCa/python/autoCond.py [hunk 1]
```diff
@@ -24,23 +24,23 @@
...
     # GlobalTag for Run2 HI data
-    'run2_data_promptlike_hi'      : '123X_dataRun2_PromptLike_HI_v3',
+    'run2_data_promptlike_hi'      : '124X_dataRun2_PromptLike_HI_v1',
     # GlobalTag with fixed snapshot time for Run2 HLT RelVals: customizations to run with fixed L1 Menu
     'run2_hlt_relval'              : '123X_dataRun2_HLT_relval_v3',
     # GlobalTag for Run3 HLT: identical to the online GT (124X_dataRun3_HLT_v1) but with snapshot at 2022-06-08 15:00:00 (UTC)
     'run3_hlt'                     : '124X_dataRun3_HLT_frozen_v1',
     # GlobalTag with fixed snapshot time for Run3 HLT RelVals: customizations to run with fixed L1 Menu
     'run3_hlt_relval'              : '124X_dataRun3_HLT_relval_v2',
     # GlobalTag for Run3 data relvals (express GT) - identical to 123X_dataRun3_Express_v8 but with snapshot at 2022-05-31 20:00:00 (UTC)
```
Comment (tvami) [inline Configuration/AlCa/python/autoCond.py:40]:
```suggestion
    # GlobalTag for Run3 data relvals (express GT) - identical to 124X_dataRun3_Express_v1 but with snapshot at 2022-06-09 20:00:00 (UTC)
```

File: Configuration/AlCa/python/autoCond.py [hunk 2]
```diff
@@ -24,23 +24,23 @@
...
-    'run3_data_express'            : '123X_dataRun3_Express_frozen_v4',
+    'run3_data_express'            : '123X_dataRun3_Express_frozen_v1',
     # GlobalTag for Run3 data relvals (prompt GT) - identical to 123X_dataRun3_Prompt_v10 but with snapshot at 2022-05-31 20:00:00 (UTC)
```
Comment (tvami) [inline Configuration/AlCa/python/autoCond.py:42]:
```suggestion
    # GlobalTag for Run3 data relvals (prompt GT) - identical to 124X_dataRun3_Prompt_v1 but with snapshot at 2022-06-09 20:00:00 (UTC)
```

**Comment by mmusich** [comment]
please test

File: Configuration/AlCa/python/autoCond.py [hunk 3]
```diff
@@ -24,23 +24,23 @@
...
     # GlobalTag for Run3 HLT: identical to the online GT (124X_dataRun3_HLT_v1) but with snapshot at 2022-06-08 15:00:00 (UTC)
     'run3_hlt'                     : '124X_dataRun3_HLT_frozen_v1',
     # GlobalTag with fixed snapshot time for Run3 HLT RelVals: customizations to run with fixed L1 Menu
     'run3_hlt_relval'              : '124X_dataRun3_HLT_relval_v2',
-    # GlobalTag for Run3 data relvals (express GT) - identical to 123X_dataRun3_Express_v8 but with snapshot at 2022-05-31 20:00:00 (UTC)
-    'run3_data_express'            : '123X_dataRun3_Express_frozen_v4',
-    # GlobalTag for Run3 data relvals (prompt GT) - identical to 123X_dataRun3_Prompt_v10 but with snapshot at 2022-05-31 20:00:00 (UTC)
-    'run3_data_prompt'             : '123X_dataRun3_Prompt_frozen_v4',
+    # GlobalTag for Run3 data relvals (express GT) - identical to 124X_dataRun3_Express_v1 but with snapshot at 2022-06-09 20:00:00 (UTC)
+    'run3_data_express'            : '123X_dataRun3_Express_frozen_v1',
```
Comment (tvami) [inline Configuration/AlCa/python/autoCond.py:41]:
```suggestion
    'run3_data_express'            : '124X_dataRun3_Express_frozen_v1',
```

File: Configuration/AlCa/python/autoCond.py [hunk 4]
```diff
@@ -24,23 +24,23 @@
...
+    # GlobalTag for Run3 data relvals (prompt GT) - identical to 124X_dataRun3_Prompt_v1 but with snapshot at 2022-06-09 20:00:00 (UTC)
+    'run3_data_prompt'             : '123X_dataRun3_Prompt_frozen_v1',
```
Comment (tvami) [inline Configuration/AlCa/python/autoCond.py:43]:
```suggestion
    'run3_data_prompt'             : '124X_dataRun3_Prompt_frozen_v1',
```

**Review by tvami — COMMENTED** [review]
Sorry I made a(n important) typo in the cmsTalk announcement, these are supposed to be 12_4_X GTs

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by tvami** [comment]
+alca
 * wf testing this development runs fine https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-edcbe3/25473/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5+ALCAHARVD7+ALCAHARVD8/
 * @dmeuser please update the PR describtion with the following
```
This PR updates the autoCond with GTs containing SiPixelAliThresholdsHG_offline_v0 and SiPixelAliThresholdsHG_express_v0

The new GTs are

124X_dataRun2_v2
124X_dataRun2_relval_v2
124X_dataRun2_HEfail_v2

124X_dataRun3_v3
124X_dataRun3_relval_v3

124X_dataRun3_Express_frozen_v1
124X_dataRun2_PromptLike_HI_v1
124X_dataRun3_Prompt_frozen_v1
where the frozen GTs are the snapshotter versions of the newly created

124X_dataRun3_Express_v1
124X_dataRun3_Prompt_v1
The snapshot time is
2022-06-09 20:00:00 (UTC)

The diffs

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_v1/124X_dataRun2_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_relval_v1/124X_dataRun2_relval_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun2_HEfail_v1/124X_dataRun2_HEfail_v2

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_v2/124X_dataRun3_v3

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/124X_dataRun3_relval_v2/124X_dataRun3_relval_v3

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun3_Express_frozen_v4/124X_dataRun3_Express_frozen_v1

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun2_PromptLike_HI_v3/124X_dataRun2_PromptLike_HI_v1

https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/123X_dataRun3_Prompt_frozen_v4/124X_dataRun3_Prompt_frozen_v1
```

**Comment by dmeuser** [comment]
>     * @dmeuser please update the PR describtion with the following

@tvami done

**Comment by qliphy** [comment]
+1

**Pull request merged**